go-controller/pkg/ovn/pods.go

```diff
@@ -107,69 +107,40 @@ func (oc *Controller) deletePodAcls(logicalPort string) {
 	}
 }
 
-func (oc *Controller) getLogicalPortUUID(logicalPort string) (string, error) {
-	if oc.logicalPortUUIDCache[logicalPort] != "" {
-		return oc.logicalPortUUIDCache[logicalPort], nil
-	}
-
-	out, stderr, err := util.RunOVNNbctl("--if-exists", "get",
-		"logical_switch_port", logicalPort, "_uuid")
-	if err != nil {
-		return "", fmt.Errorf("Error while getting uuid for logical_switch_port "+
-			"%s, stderr: %q, err: %v", logicalPort, stderr, err)
-	}
-
-	if out == "" {
-		return "", fmt.Errorf("empty uuid for logical_switch_port %s", logicalPort)
-	}
-
-	oc.logicalPortUUIDCache[logicalPort] = out
-	return oc.logicalPortUUIDCache[logicalPort], nil
-}
-
 func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 	if pod.Spec.HostNetwork {
 		return
 	}
 
 	podDesc := pod.Namespace + "/" + pod.Name
 	klog.Infof("Deleting pod: %s", podDesc)
-	logicalPort := podLogicalPortName(pod)
-	out, stderr, err := util.RunOVNNbctl("--if-exists", "lsp-del",
-		logicalPort)
-	if err != nil {
-		klog.Errorf("Error in deleting pod %s logical port "+
-			"stdout: %q, stderr: %q, (%v)",
-			podDesc, out, stderr, err)
-	}
 
-	var podIP net.IP
-	podAnnotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
+	logicalPort := podLogicalPortName(pod)
+	portInfo, err := oc.logicalPortCache.get(logicalPort)
 	if err != nil {
-		klog.V(5).Infof("failed to read pod %s annotation when deleting "+
-			"logical port; falling back to PodIP %s: %v",
-			podDesc, pod.Status.PodIP, err)
-		podIP = net.ParseIP(pod.Status.PodIP)
-	} else {
-		podIP = podAnnotation.IP.IP
+		klog.Errorf(err.Error())
+		return
 	}
 
-	delete(oc.logicalPortCache, logicalPort)
-
-	oc.lspMutex.Lock()
-	delete(oc.logicalPortUUIDCache, logicalPort)
-	oc.lspMutex.Unlock()
-
 	// Remove the port from the default deny multicast policy
 	if oc.multicastSupport {
-		if err := oc.podDeleteDefaultDenyMulticastPolicy(logicalPort); err != nil {
+		if err := podDeleteDefaultDenyMulticastPolicy(portInfo); err != nil {
 			klog.Errorf(err.Error())
 		}
 	}
 
-	if err := oc.deletePodFromNamespace(pod.Namespace, podIP, logicalPort); err != nil {
+	if err := oc.deletePodFromNamespace(pod.Namespace, portInfo); err != nil {
 		klog.Errorf(err.Error())
 	}
+
+	out, stderr, err := util.RunOVNNbctl("--if-exists", "lsp-del", logicalPort)
```

Comment:
Do we have a retry function (like RunOVNNbctlWithRetry) to attempt to delete multiple times in case of a failure? The only way I see this failing is if there is some connectivity problem. Since we remove from the cache right after this, we would lose its context and the port would be left dangling in OVN.

Reply:
@trozet there are several places in the code where we might have stale info in OVN. We are working on a sync thread or go routine that wakes up every interval and checks for stale OVN data and cleans it up just like the sync*() functions we have in the code. That thread also will check for resources that are in K8s but not in OVN and tries to flag such cases and/or fixes it.

Reply:
thanks